src/overlays/actors/ovl_En_Test4/z_en_test4.c

```diff
@@ -24,28 +31,558 @@ const ActorInit En_Test4_InitVars = {
+            this->nextBellTime = CLOCK_TIME(17, 54);
+        } else {
+            this->nextBellTime = CLOCK_TIME(5, 30);
+        }
+    } else {
+        if (gSaveContext.time < CLOCK_TIME(5, 30)) {
+            this->nextBellTime = CLOCK_TIME(5, 30);
+        } else if (gSaveContext.time < CLOCK_TIME(5, 36)) {
+            this->nextBellTime = CLOCK_TIME(5, 36);
+        } else if (gSaveContext.time < CLOCK_TIME(5, 42)) {
+            this->nextBellTime = CLOCK_TIME(5, 42);
+            gSaveContext.unk_3F64 -= 50.0f;
+        } else if (gSaveContext.time < CLOCK_TIME(5, 48)) {
+            this->nextBellTime = CLOCK_TIME(5, 48);
+            gSaveContext.unk_3F64 -= 100.0f;
+        } else if (gSaveContext.time < CLOCK_TIME(5, 54)) {
+            this->nextBellTime = CLOCK_TIME(5, 54);
+            gSaveContext.unk_3F64 -= 150.0f;
+        } else if (gSaveContext.time < CLOCK_TIME(6, 0)) {
+            this->nextBellTime = CLOCK_TIME(17, 30);
+            gSaveContext.unk_3F64 -= 200.0f;
+        } else {
+            this->nextBellTime = CLOCK_TIME(17, 30);
+        }
 
+        if ((sCutscenes[this->unk_144] < 0) || ((globalCtx->actorCtx.unk5 & 2) != 0) || (CURRENT_DAY == 3) ||
+            (gSaveContext.time >= CLOCK_TIME(17, 0))) {
+            gSaveContext.unk_3F64 = 1000.0f;
+        }
+        if (gSaveContext.unk_3F64 != 1000.0f) {
+            gSaveContext.unk_3F60 = 1;
+        }
+    }
+}
+
+#ifdef NON_MATCHING
+// 1 instruction in the wrong place
+void EnTest4_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnTest4* this = THIS;
+    Player* player = PLAYER;
+    s8 temp_v0 = this->actor.cutscene;
+
+    sCutscenes[0] = temp_v0;
+    if (temp_v0 >= 0) {
+        ActorCutscene* temp_v0_2 = ActorCutscene_GetCutscene(sCutscenes[0]);
+
+        gSaveContext.eventInf[5] |= 0x4;
+        sCutscenes[1] = temp_v0_2->additionalCutscene;
+    } else {
+        gSaveContext.eventInf[5] &= (u8)~0x4;
+        sCutscenes[1] = sCutscenes[0];
+    }
+
+    if (sIsLoaded || (gSaveContext.eventInf[2] & 0x80)) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        sIsLoaded = true;
+        this->actor.room = -1;
+        gSaveContext.unk_3F60 = 0;
+        gSaveContext.unk_3F64 = 1000.0f;
+        if (CURRENT_DAY == 0) {
+            if (gSaveContext.time < CLOCK_TIME(6, 1)) {
+                gSaveContext.time = CLOCK_TIME(6, 0);
+                gSaveContext.gameMode = 0;
+                do {
+                    GameState* state = &globalCtx->state;
+                    state->running = false;
+                } while (0);
+                SET_NEXT_GAMESTATE(&globalCtx->state, Daytelop_Init, DaytelopContext);
+                this->unk_144 = 1;
+                gSaveContext.time = CLOCK_TIME(6, 0);
+                Actor_MarkForDeath(&this->actor);
+            } else {
+                gSaveContext.day = 1;
+                pad = gSaveContext.day;
+                gSaveContext.daysElapsed = pad;
```

Comment:
Since `pad` *is* used, I think it should be named something different. Maybe something like `dayTemp`?

Reply:
Oh, yeah, my bad